cohortextractor/dsl_example.py

```diff
@@ -0,0 +1,51 @@
+from cohortextractor.dsl import (
+    Codelist,
+    Cohort,
+    Column,
+    EventTable,
+    PatientTable,
+    categorise,
+)
+
+
+# These table definitions will be imported from elsewhere, and researchers will not have
+# to write them.
+
+
+class Patients(PatientTable):
+    patient_id = Column("patient_id")
+    sex = Column("sex")
+
+
+class Immunisations(EventTable):
+    patient_id = Column("patient_id")
+    date = Column("date")
+    code = Column("code")
+
+
+patients = Patients()
+imms = Immunisations()
+
+# This is a cohort definition, of the kind that researchers will have to implement.
+
+cohort = Cohort()
+cohort.sex = patients.select_column(patients.sex)
```

Comment:
Is it confusing that we'll presumably import `Patients` and `Immunisations` (or whatever) from the same place, but e.e. `select_column` will return a different thing from each of them?  I guess it will return a `PatientSeries` from `patients`, and error if you tried to call it on `imms` without sorting/limiting first.

Maybe it's not, and it's obvious that the patients table doesn't need reducing and others do

Reply:
It might be confusing, but we can mitigate against it by catching the lookup in `.__getattr__` and providing a good error message.